### ISSUE TITLE
eslint (2.0) erroneously complains about correct spacing

### ISSUE THREAD

**Issue by WhyEvenTry** [eslint/eslint#5237]
Test files (renamed to have ending .txt for github to accept them as issue attachment):

[test.js.txt](https://github.com/eslint/eslint/files/129084/test.js.txt)
[.eslintrc.js.txt](https://github.com/eslint/eslint/files/129085/default.eslintrc.js.txt)

The code 

```
const runImapServer = function (imapAccountSettingsHash) {
    return Promise.resolve()
    .then(() => {
        [
            'APPEND', 'CREATE', 'RENAME', 'COPY', 'UID COPY', 'EXPUNGE', 'STORE', 'DELETE'
        ].forEach(command => {
            // ...
        });
    });
};
```

causes warnings/errors of the type

    Expected indentation of 12 space characters but found 8   indent

for the lines containing the (strings) array. **However**, not only is the spacing correct, what's even more telling that this is a bug is that if I change the function expression from being assigned to a `const` to instead assigning it to `module.exports.someVar` the complaining stops and the spaces are accepted! See the attached file for the full example.

This is what I get when I run `eslint test.js`:
```

$ eslint test.js

C:\Users\Michael\OneDrive\Projects\One\test.js
  6:9   warning  Expected indentation of 12 space characters but found 8   indent
  7:13  warning  Expected indentation of 16 space characters but found 12  indent
  8:9   warning  Expected indentation of 12 space characters but found 8   indent

✖ 3 problems (0 errors, 3 warnings)
```

**Comment by BYK** [comment]
@gyandeeps - looks like a bug to me. What do you think?

@WhyEvenTry - according to your eslintrc file looks like you are using ESLint 2.0.0 but can you provide us the specific version you are using (beta/rc etc. or release?) and Node version you are on?

**Comment by WhyEvenTry** [comment]
v2.0.0, node.js 5.6.0

**Comment by BYK** [comment]
Hey, thanks for providing the info. I tried adding the following test case to `indent` rule and it passes:

```js
        {
            code:
            "module.exports.func = function (opts) {\n" +
            "    return Promise.resolve()\n" +
            "    .then(() => {\n" +
            "        [\n" +
            "            'ONE', 'TWO'\n" +
            "        ].forEach(command => { doSomething(); });\n" +
            "    });\n" +
            "};",
            parserOptions: { ecmaVersion: 6 },
            options: [4, {"SwitchCase": 1}]
        }
```

**Comment by BYK** [comment]
@WhyEvenTry - anything jumps out from my isolated test case that is different than your code?

**Comment by WhyEvenTry** [comment]
Yes, you test with `module.exports` - which is exactly the one of the two cases of my submitted test cases that passes.

**Comment by BYK** [comment]
@WhyEvenTry - lol, I thought the reverse. Anyways, I was able to reproduce this with `const func =`. Looks like it has something to do with `VariableDeclarator` option since setting that to `0` removes the error.

I'll see if I can fix this.

**Comment by WhyEvenTry** [comment]
You guys are fast!

**Comment by feross** [comment]
This fix may have introduced a new bug: https://github.com/eslint/eslint/issues/5295

**Comment by BYK** [comment]
@feross it indeed did. Submitted a fix to fix the fix :P
